GitCommands/RevisionReader.cs

```diff
@@ -37,19 +39,18 @@ public sealed class RevisionReader
         private const int _offsetDaysForOldestBody = 6 * 30; // about 6 months
 
         // reflog selector to identify stashes
-        private readonly bool _hasReflogSelector;
+        private bool _hasReflogSelector;
 
-        private string LogFormat => _hasReflogSelector ? FullFormat.Replace("%B", "%gD%n%B") : FullFormat;
+        private string LogFormat => string.Format(_fullFormat, _hasReflogSelector ? _reflogSelectorFormat : "");
```

Comment:
❔ Replace `_hasReflogSelector` but specify it as argument:

```suggestion
        private string LogFormat(bool hasRefLogSelector) => string.Format(_fullFormat, hasReflogSelector ? _reflogSelectorFormat : "");
```

Perhaps with a default argument value if it is always the same for `BuildArguments()`.

Reply:
Changing so _hasReflogSelector  is set in this method, it must match when parsing the logs.
Set default to false (changed my mind, why force pushed)